gcsio/src/test/java/com/google/cloud/hadoop/gcsio/TrackingHttpRequestInitializer.java

```diff
@@ -169,12 +171,33 @@ public ImmutableList<String> getAllRequestStrings() {
         .collect(toImmutableList());
   }
 
+  public ImmutableList<String> getAllRequestInvocationIds() {
+    return requests.stream()
+        .map(r -> getInvocationId(r.getHeaders()))
+        .filter(id -> id != null && !id.isEmpty())
+        .collect(toImmutableList());
+  }
+
   public ImmutableList<String> getAllRawRequestStrings() {
     return requests.stream()
         .map(GoogleCloudStorageIntegrationHelper::requestToString)
         .collect(toImmutableList());
   }
 
+  private String getInvocationId(HttpHeaders header) {
+    String apiClientHeader = (String) header.get(InvocationIdInterceptor.GOOG_API_CLIENT);
```

Comment:
can this be null?

Reply:
I don't expect the  httpHeader to be `null` for any request.

Reply:
I think it is not in our control and better to be defensive in this case.

Reply:
Ignore this comment since this is for test code.